tests/multi_gcov_aggregator.py

```diff
@@ -0,0 +1,145 @@
+                match = re.match(r'^\s*(\-|#####|\d+\*?|\d+):\s*(\d+):\s*(.*)', line)
+                if match:
+                    count, line_number, line_content = match.groups()
+                    # get the line of code, preserving spaces
+                    i_first_colon = line.find(':')
+                    i_second_colon = line[i_first_colon+1:].find(':')
+                    line_content = line[i_first_colon+i_second_colon+2:-1]
+
+                    if (count == "-") or (count == "#####"):  # Line not executable or not executed
+                        stripped_line = line_content.strip()
+                        keywords = ("integer", "logical", "character", "real","complex", "type",
+                                    "INTEGER", "LOGICAL", "CHARACTER", "REAL","COMPLEX", "TYPE",
+                                    "module",  "private", "public", "contains", "interface",
+                                    "MODULE",  "PRIVATE", "PUBLIC", "CONTAINS", "INTERFACE",
+                                    "use", "implicit", "namelist",
+                                    "USE", "IMPLICIT", "NAMELIST",
+                                    "else", "end", "case", "continue",
+                                    "ELSE", "END", "CASE", "CONTINUE",
+                                    "!", "#", "&",
+                                    "1","2","3","4","5","6","7","8","9") #format strings
+                        if len(stripped_line) == 0 or stripped_line.startswith(keywords):
+                            count = "-"  # Line is never executable
+                        else:
+                            count = 0    # Line is executable but not covered, possibly due to preprocessor directives
+                    elif count.endswith("*"):  # Branch execution
+                        count = int(count[:-1])
+                    else:
+                        count = int(count)
+
+                    line_number = int(line_number)
+                    # Aggregate counts and keep line content
+                    current_count, current_line_content, directories = self.coverage_data[source_file][line_number]
+
+                    if isinstance(count, int) and count > 0 and directory is not None:
+                        directories.append(directory)
+
+                    if isinstance(count, int):
+                        if isinstance(current_count, int):
+                           current_count += count
+                        else:
+                           current_count = count
+
+                    if line_content != current_line_content and current_line_content != "":
+                        print(f"Warning: Line content mismatch in {source_file} at line {line_number}")
+                        print(f"  Previous line content: {current_line_content}")
+                        print(f"  Current line content: {line_content}")
+                        print(f"  Using previous line content and count for aggregation.")
```

Comment:
This should go in the error not in stdout.
```suggestion
                        print(f"Warning: Line content mismatch in {source_file} at line {line_number}", file=sys.stderr)
                        print(f"  Previous line content: {current_line_content}", file=sys.stderr)
                        print(f"  Current line content: {line_content}", file=sys.stderr)
                        print("  Using previous line content and count for aggregation.", file=sys.stderr)
```
Note: the suggestion assumes `sys` has been imported